march_gain_scheduling/src/march_gain_scheduling/one_step_linear_interpolation.py

```
@@ -0,0 +1,21 @@
+#!/usr/bin/env python
+
+import rospy
+
+
+def interpolate(current_gains, needed_gains, gradient, delta_t):
+    if len(current_gains) != len(needed_gains):
+        rospy.logerr("needed_gains and current_gains not equal length")
+        return current_gains
```

It is better to just throw an exception here. When this happens there is a significant problem with how this function is used. In the code this function is used, one can catch this exception (i.e. with try and except https://www.pythonforbeginners.com/error-handling/python-try-and-except/) and prevent the exo from shutting down. Its probably best to make your own exception (@Olavhaasie?) see https://www.programiz.com/python-programming/user-defined-exception

---

Probably best to define your own

---

I am trying to implement this into the code, but I can't figure out how.
This was my first guess:
`try:
        for i in range(len(current_gains)):
            if current_gains[i] > needed_gains[i]:
                 next_gains[i] = max(needed_gains[i], current_gains[i] - gradient * delta_t)
            else:
                 next_gains[i] = min(needed_gains[i], current_gains[i] - gradient * delta_t)
except:
           ....  
`
The problem when using this implementation is that you cover the situation where current_gains is larger than needed_gains, but not the other way around. Because that implementation will not make the loop fail.. Or do I need to make a implementation like this: